backend/tracim_backend/lib/core/user.py

```diff
@@ -1230,3 +1244,117 @@ def get_about_user(self, user_id: int) -> AboutUser:
+        _, original_file_extension = os.path.splitext(self._user.cropped_avatar.filename)
+        return StorageLib(self._config).get_jpeg_preview(
+            depot_file=user.cropped_avatar,
+            filename=filename,
+            default_filename=default_filename,
+            width=width,
+            height=height,
+            original_file_extension=original_file_extension,
+            force_download=force_download,
+            page_number=1,
+        )
+
+    def set_avatar(
+        self, user_id: int, new_filename: str, new_mimetype: str, new_content: typing.BinaryIO
+    ) -> None:
+        user = self.get_one(user_id)
+        label, file_extension = os.path.splitext(new_filename)
+        with tempfile.SpooledTemporaryFile(mode="wb+") as tmp_file:
+            tmp_file.write(new_content.read())
+            tmp_file.seek(0, 0)
+            user.avatar = FileIntent(tmp_file.read(), new_filename, new_mimetype)
+            tmp_file.seek(0, 0)
+            with tempfile.SpooledTemporaryFile(mode="wb+") as cropped_tmp_file:
+                # FIXME - G.M - 2021-01-21 - should we catch error here,
+                # what happened if pillow failed ?
+                crop_image(tmp_file, cropped_tmp_file, ratio=AVATAR_RATIO)
+                cropped_tmp_file.seek(0, 0)
+                user.cropped_avatar = FileIntent(
+                    cropped_tmp_file.read(), new_filename, new_mimetype
+                )
+            self._session.add(user)
+            self._session.flush()
+
+    def get_cover(
+        self, user_id: int, filename: str, default_filename: str, force_download: bool = False,
+    ) -> HapicFile:
+        user = self.get_one(user_id)
+        if not user.cover:
+            raise UserCoverNotFound("cover of user {} not found".format(user_id))
+        return StorageLib(self._config).get_raw_file(
+            depot_file=user.cover,
+            filename=filename,
+            default_filename=default_filename,
+            force_download=force_download,
+        )
+
+    def get_cover_preview(
+        self,
+        user_id: int,
+        filename: str,
+        default_filename: str,
+        width: int = None,
+        height: int = None,
+        force_download: bool = False,
+    ) -> HapicFile:
+        user = self.get_one(user_id)
+        if not user.cropped_cover:
+            raise UserCoverNotFound("cropped version of user {} cover not found".format(user_id))
+        _, original_file_extension = os.path.splitext(self._user.cropped_cover.filename)
+        return StorageLib(self._config).get_jpeg_preview(
+            depot_file=user.cropped_cover,
+            filename=filename,
+            default_filename=default_filename,
+            width=width,
+            height=height,
+            original_file_extension=original_file_extension,
+            force_download=force_download,
+            page_number=1,
+        )
+
+    def set_cover(
```

Comment:
Shouldn't this code be shared with `set_avatar`?

Reply:
This one could effectively.

Reply:
humm... finally not really sure about this one. it's definitively both same code but don't see easy way to split them.

Reply:
Proposal as a diff against your file: 
[cover_avatar.txt](https://github.com/tracim/tracim/files/5860463/cover_avatar.txt)